private/buf/bufgen/migrate.go

```diff
@@ -0,0 +1,553 @@
+				FileOption: bufgenv2.FileOptionGoPackagePrefix.String(),
+				Module:     module.IdentityString(),
+				Value:      override,
+			}
+			managedConfigV2.Override = append(managedConfigV2.Override, moduleOverrideRule)
+		}
+	}
+	if objcClassPrefixConfig := managedConfigV1.ObjcClassPrefixConfig; objcClassPrefixConfig != nil {
+		defaultOverrideRule := bufgenv2.ExternalManagedOverrideConfigV2{
+			FileOption: bufgenv2.FileOptionObjcClassPrefix.String(),
+			Value:      objcClassPrefixConfig.Default,
+		}
+		managedConfigV2.Override = append(managedConfigV2.Override, defaultOverrideRule)
+		for _, excludedModule := range objcClassPrefixConfig.Except {
+			moduleDisableRule := bufgenv2.ExternalManagedDisableConfigV2{
+				FileOption: bufgenv2.FileOptionObjcClassPrefix.String(),
+				Module:     excludedModule.IdentityString(),
+			}
+			managedConfigV2.Disable = append(managedConfigV2.Disable, moduleDisableRule)
+		}
+		for module, override := range objcClassPrefixConfig.Override {
+			moduleOverrideRule := bufgenv2.ExternalManagedOverrideConfigV2{
+				FileOption: bufgenv2.FileOptionObjcClassPrefix.String(),
+				Module:     module.IdentityString(),
+				Value:      override,
+			}
+			managedConfigV2.Override = append(managedConfigV2.Override, moduleOverrideRule)
+		}
+	}
+	if rubyPackageConfig := managedConfigV1.RubyPackageConfig; rubyPackageConfig != nil {
+		for _, excludedModule := range rubyPackageConfig.Except {
+			moduleDisableRule := bufgenv2.ExternalManagedDisableConfigV2{
+				FileOption: bufgenv2.FileOptionRubyPackage.String(),
+				Module:     excludedModule.IdentityString(),
+			}
+			managedConfigV2.Disable = append(managedConfigV2.Disable, moduleDisableRule)
+		}
+		for module, override := range rubyPackageConfig.Override {
+			moduleOverrideRule := bufgenv2.ExternalManagedOverrideConfigV2{
+				FileOption: bufgenv2.FileOptionRubyPackage.String(),
+				Module:     module.IdentityString(),
+				Value:      override,
+			}
+			managedConfigV2.Override = append(managedConfigV2.Override, moduleOverrideRule)
+		}
+	}
+	for fileOption, fileToOverride := range managedConfigV1.Override {
+		for file, override := range fileToOverride {
+			fileOverrideRule := bufgenv2.ExternalManagedOverrideConfigV2{
+				FileOption: strings.ToLower(fileOption),
+				Path:       file,
+				Value:      override,
+			}
+			managedConfigV2.Override = append(managedConfigV2.Override, fileOverrideRule)
+		}
+	}
+	return &managedConfigV2
+}
+
+func getExternalInputConfigV2(
+	ctx context.Context,
+	logger *zap.Logger,
+	input string,
+	types []string,
+	includePaths []string,
+	excludedPaths []string,
+) (*bufgenv2.ExternalInputConfigV2, error) {
+	inputConfig := bufgenv2.ExternalInputConfigV2{}
+	path, options, err := bufref.GetRawPathAndOptions(input)
+	if err != nil {
+		return nil, err
+	}
+	format, err := buffetch.NewRefParser(logger).GetRefFormat(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+	switch format {
+	case buffetch.FormatBin:
```

Comment:
This (and the below) feels like an abuse of the `buffetch` API  boundary - it feels like all of this stuff should somehow live at the buffetch layer. I don't know, I don't have any great answers to be honest.